Design.md

```diff
@@ -32,71 +32,73 @@ Here are the IELE design decisions described from the perspective of differences
-* change the RETURN and REVERT instructions to take an arbitrary number of register operands. REVERT reverts the contract, but RETURN will return to the instruction position on the top of the local call stack if one exists, restoring the values of local registers and copying the return values into the return value operands of the call instruction. If the local call stack is empty, it returns from the contract.
-* change the CALL, CALLCODE, DELEGATECALL, and STATICCALL instructions to take an arbitrary number of register arguments and return values instead of memory ranges. The arguments are copied into local registers 0 to N-1 of the function being called. When the contract returns, the returned values are copied into the return value operands of the call instruction.
-* the CALL, CALLCODE, DELEGATECALL, and STATICCALL instructions also take an immediate data segment offset. The contract is malformed if the offset does not refer to a FUNCTION instruction. The instruction contains the name of the function being called.
-* if the name of the function being called is not in the data segment for the contract being called, it is an exception. If the name of the function is in the data segment but the function is not exported as external, it is also an exception. Otherwise, the called contract begins at the start of the function whose function label is the data segment offset of the name of the function in the called contract.
-* In all cases, if a mismatch occurs between the number of arguments or return values of a function and the matching CALL or RETURN instruction, an exception occurs.
-* Because it is no longer needed, we remove the CALLDATA* instructions and the RETURNDATA* instructions.
-* Because the function being called is now explicitly part of the signature of calling a contract, we condense the 7 remaining precompiled contracts into a single contract with seven functions.
+* IELE has a `call` instruction used for local calls to other functions within the contract. `call` takes a function name, an arbitrary number of register arguments, and an arbitrary number of register return values, and jumps to the start of the function in the current contract with the corresponding name. It pushes the current instruction position, the values of all local registers, and the register operands of the return values onto the local call stack. The arguments of the call are copied into local registers corresponding to the formal arguments of the function being called.
+* The `ret` and `revert` instructions take an arbitrary number of register operands. `revert` reverts the contract, but `ret` will return to the instruction position on the top of the local call stack if one exists, restoring the values of local registers and copying the return values into the return value operands of the call instruction. If the local call stack is empty, it returns from the contract.
```

Comment:
How does the caller receive the values if revert is called with multiple arguments?

Reply:
Actually both revert and return return values the same way, I will fix the text to reflect that.